packages/frontend/cypress/e2e/login.cy.ts

```diff
@@ -22,12 +22,13 @@ describe('sign up, log out, then sign in', () => {
         cy.findByText('Sign out').click()
         cy.findByText('Get started').click()
         cy.findByText('Sign In').click()
-        cy.get('@iden').then((iden) => {
-            cy.get('textarea').type(iden, {
-                parseSpecialCharSequences: false,
-            })
-        })
-        cy.get('#signin').click()
+
+        // cy.get('@iden').then((iden) => {
```

Comment:
Why are these tests commented? We definitely want to e2e test logging in

Reply:
Made comments because was not sure if that could be used to test the sign in with password. Not exactly sure how to get the private key for the sign in flow.

Reply:
pretty sure the solution to test login is that the private key input needs to be formatted correctly -- but what is the exact format? Testing locally, I get parsing errors in the console when  typing and submitting the private key input and feel like I'm missing something obvious here @vimwitch

Reply:
Hmm, if you mean for the encrypted version it should be like `{ data: string, iv: string }`. The way I was doing e2e tests was running a signup, saving the key, then doing a signin with the saved key. That way we avoid needing to know/care about the formatting. As long as the webapp can parse it's own output it seems okay.

On a separate note the formatting for encrypted files could be better. We should store info about the cypher/operation mode like ethereum keystores do. We could fork [Keythereum](https://www.npmjs.com/package/keythereum) and make a version compatible with semaphore identities. The only difference is we need 64 bytes of random data for a key, instead of 32.

We could also run an Ethereum key through sha512 and then we'd have 64 bytes, but I'm not sure if reducing the key security to 256 bits is acceptable.

Reply:
Might also be better to do the Keythereum fork in another PR because I need to gain more familiarity with Semaphore.

Reply:
No, this change would be entirely inside the webapp. The tests don't need to know about how we encrypt the keys. And yes, definitely something for a different pr.

The `/api/signup` endpoint doesn't use encryption. It always takes the identity commitment. None of the tests should worry about the specific of encryption, they should just test that the encrypted signup/logic paths work.